src/App.tsx

```diff
@@ -1,26 +1,219 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { HeaderForm } from './components/HeaderForm/HeaderForm';
+import { Footer } from './components/Footer/Footer';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [newTodoTitle, setTodoTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [filteredBy, setFilteredBy] = useState(Filter.All);
+  const [todosCounter, setTodosCounter] = useState(0);
+  const shouldRenderFooter = todos && todos.length > 0;
+  const completedTodos = todos.filter(todo => todo.completed);
+  const completedTodosIds = completedTodos.map(todo => todo.id);
+
+  const loadTodos = useCallback(() => {
+    setLoading(true);
+    todoService
+      .getTodos()
+      .then(fetchedTodo => {
+        setTodos(fetchedTodo);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setShowError(true);
+        setTimeout(() => setShowError(false), 3000);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (filteredBy) {
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filteredBy]);
+
+  function addTodo(newTodo: Todo) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const addTodo = (newTodo: Todo) => {
```